beacon_node/http_api/src/lib.rs

```diff
@@ -1487,29 +1487,45 @@ pub fn serve<T: BeaconChainTypes>(
         .and(warp::path("beacon"))
         .and(warp::path("blob_sidecars"))
         .and(block_id_or_err)
+        .and(warp::query::<api_types::BlobIndicesQuery>())
         .and(warp::path::end())
         .and(chain_filter.clone())
         .and(warp::header::optional::<api_types::Accept>("accept"))
+
         .and_then(
             |block_id: BlockId,
+             indices: api_types::BlobIndicesQuery,
              chain: Arc<BeaconChain<T>>,
              accept_header: Option<api_types::Accept>| {
                 async move {
-                    let blob_sidecar_list = block_id.blob_sidecar_list(&chain).await?;
-
+                    let blob_sidecar_list_indexed = block_id.blob_sidecar_list_indexed(indices, &chain).await?;
+                    // let blob_sidecar_list_indexed = match indices.indices {
+                    //     Some(vec) => {
+                    //         let list = blob_sidecar_list.into_iter()
+                    //         .enumerate()
+                    //         .filter(|(index, _)| vec.contains(index))
+                    //         .map(|(_, blob)| blob)
+                    //         .collect();
+                    //         let indexed_list = BlobSidecarList::new(list);
+                    //         indexed_list.unwrap()
+                    //     },
+                    //     None => {
+                    //         blob_sidecar_list
+                    //     },
+                    // };    
```

Comment:
Please remove commented code.

Just a minor comment on naming of `blob_sidecar_list_indexed`, I think it's slightly confusing because it's just a list (potentially filtered), maybe `blob_sidecar_list_filtered` or something similar would be clearer?